### ISSUE TITLE
Can MAGIC be used in more complex grf scenarios? 

### ISSUE THREAD

**Issue by GP413413** [CORE-Robotics-Lab/MAGIC#6]
I have run MAGIC and other baselines successfully and their results are excellent. I wonder if MAGIC can be used in more complex google research football environments, like 11 vs 11 full game? What should I do and where need to make modifications? Please give me some advice and thank you for your help :)

**Comment by chrisyrniu** [comment]
Thanks for your interests in our work! Yes, MAGIC could be used in more complex environments like Google Football with more players, and the graph generation module could be more valuable when there are more agents. You can check available scenarios in GFootball of my fork [here](https://github.com/chrisyrniu/football/tree/master/gfootball/scenarios), then change the scenario name of your choice in the script line:
https://github.com/CORE-Robotics-Lab/MAGIC/blob/0ad3a6126f46e475f8d46ab61e67fddbfe99e7d8/train_grf.sh#L28
and the number of agents in the script line:
https://github.com/CORE-Robotics-Lab/MAGIC/blob/0ad3a6126f46e475f8d46ab61e67fddbfe99e7d8/train_grf.sh#L6

Meanwhile, you might need to change hyperparameters like the batch size and the learning rate to get it work best for different environments. As a reminder, MAGIC in this repo is based on a multi-agent A2C architecture. The performance of MAGIC could be potentially improved with more recent MARL frameworks like [MAPPO](https://github.com/marlbenchmark/on-policy).

**Comment by GP413413** [comment]
Thank you for your reply! As your comment, I have changed the scenario name into 11_vs_11_easy_stochastic, nagents as well as num_controlled_lagents to 10. Moreover, I have also changed max_steps to 600 and reward_type to scoring,checkpoints in [MAGIC/train_grf.sh](https://github.com/CORE-Robotics-Lab/MAGIC/blob/0ad3a6126f46e475f8d46ab61e67fddbfe99e7d8/train_grf.sh#L6) for training without other changes.
However, as epoch increases, the reward can change from -0.86 into +0.2~0.5, but the success rate is always 0.0. The current result  is:
Epoch 650
Reward: [0.46 0.54 0.39 0.43 0.37 0.35 0.49 0.36 0.37 0.31]
Success: 0.0000
Steps-Taken: 600.00
I wonder if the original google football environment supports multi-agent cooperative with number=10, or the calculation of success rate in complex full game scenario like 11_vs_11_stochastic may need to be modified.
I hope you can give me some advice. I will also try to change hyperparameters, for example, decreasing the batch size and the learning rate. Thank you for your help :)

**Comment by chrisyrniu** [comment]
Hi, it looks like that you changed the reward_type from "scoring" to "checkpoints," so it would be possible that the rewards for different players are different while success rate maintains zero.

The shaped checkpoints reward was experimentally revised for 3v2 (see my [commit history](https://github.com/chrisyrniu/football/commits/master/gfootball/env/wrappers.py) :)), but we did not officially evaluate any algorithms with the checkpoints reward. Feel free to try your own shaped rewards or the scoring reward! I think it would be also helpful to visualize your trained policy and keep increasing the batch size for 11 vs. 11 full games.

As a reference, more recent performing communication-based MARL methods like [multi-agent Transformer](https://github.com/PKU-MARL/Multi-Agent-Transformer) (NeurIPS 2022) and [CommFormer](https://github.com/charleshsc/CommFormer) (ICLR 2024) also demonstrated some results in GFootball including counter attack (easy) or 3 vs. 2 but not 11 vs. 11 full games yet. To begin, you might try counter-attack scenarios as a sanity check in a setting that closely resembles a full game.

Best,
Yaru

**Comment by GP413413** [comment]
Thank you for your advice. I have studied the recent algorithms MAT and CommFormer, as well as MAPPO and HAPPO. You have reminded me that MAGIC in this repo is based on a multi-agent **A2C** architecture. if I want to rewrite MAGIC and other baselines including Gacomm, TarMac, Commnet into the Actor-Critic architecture, can you think it is feasible or give me some advice? Thank you for your help :)

**Comment by chrisyrniu** [comment]
Thanks for your interests in porting our method and other baselines to more recent MARL frameworks. Please open another issue if you want to discuss technical details or problems related to MAGIC and baselines in this repo, or you can direct an email to me if you would like to discuss further.
